src/components/H2.astro

```diff
@@ -1,8 +1,12 @@
 ---
+import AnchorLink from './icons/AnchorLink.vue';
 const { id } = Astro.props;
 ---
 
-<div class="not-prose">
+<div class="not-prose group flex items-center relative">
+  <a href=`#${id}` class="px-1.5 absolute right-full opacity-0 group-hover:opacity-100">
```

Comment:
TIL absolutely positioning an element 100% to the right of its parent will put it directly to the left of it. Having it directly to the left with no gap means the anchor link won't disappear when you try to hover over it